pyg_lib/sampler/__init__.py

```diff
@@ -4,6 +4,41 @@
 from torch import Tensor
 
 
+def neighbor_sample(
+    rowptr: Tensor, col: Tensor, seed: Tensor, num_neighbors: list[int],
+    replace: bool = False, directed: bool = True, isolated: bool = True,
+    return_edge_id: bool = True
+) -> Tuple[Tensor, Tensor, Tensor, Optional[Tensor]]:
+    r"""Recursively samples neighbors from all nodes indices in :obj:`seed`
+    in the graph given by :obj:`(rowptr, col)`.
+
+    Args:
+        rowptr (torch.Tensor): Compressed source node indices.
+        col (torch.Tensor): Target node indices.
+        seed (torch.Tensor):
+        num_neighbors (list[int]): The number of neighbors to sample for each
+            node in each iteration. In heterogeneous graphs, may also take in a
+            dictionary denoting the amount of neighbors to sample for each
+            individual edge type. If an entry is set to :obj:`-1`, all neighbors
+            will be included.
+        replace (bool, optional): If set to :obj:`True`, will sample with
+            replacement. (default: :obj:`False`)
+        directed (bool, optional): If set to :obj:`False`, will include all
+            edges between all sampled nodes. (default: :obj:`True`)
+        isolated (bool, optional): #TODO(kgajdamo): add description
```

Comment:
Let's rename to `disjoint`: "If set to :obj:`True`, will create disjoint subgraphs for every seed node. (default: :obj:`False`)

Reply:
Done.